packages/runtime/src/next/tests/client.get-component-snapshot.test.ts

```diff
@@ -0,0 +1,237 @@
+import { Makeswift, MakeswiftComponentDocument } from '../client'
+import { http, HttpResponse, graphql } from 'msw'
+
+import { server } from '../../mocks/server'
+import { MakeswiftSiteVersion } from '../preview-mode'
+import { ZodError } from 'zod'
+
+const TEST_API_KEY = 'myApiKey'
+const apiOrigin = 'https://api.fakeswift.com'
+const baseUrl = `${apiOrigin}/v1/element-trees`
+
+describe('getComponentSnapshot using v1 element tree endpoint', () => {
+  test('return fallback document on 404', async () => {
+    // Arrange
+    const client = new Makeswift(TEST_API_KEY, { apiOrigin })
+    const treeId = 'myTree'
+    server.use(
+      http.get(`${baseUrl}/${treeId}`, () => HttpResponse.text('', { status: 404 }), {
+        once: true,
+      }),
+    )
+
+    // Act
+    const result = await client.getComponentSnapshot(treeId, {
+      siteVersion: MakeswiftSiteVersion.Working,
+    })
+
+    // Assert
+    expect(result.document).not.toBeNull()
+    expect(result.document.id).toBe(treeId)
+    expect(result.document.data).toBeNull()
+    expect(result.cacheData).not.toBeNull()
+    expect(result.cacheData.apiResources).toStrictEqual({})
+    expect(result.cacheData.localizedResourcesMap).toStrictEqual({})
```

Comment:
This feels like it's testing implementation details too much. I think we should write these tests as if we were consumers of `@makeswift/runtime`, based on what we can observe.